src/service.py

```diff
@@ -46,63 +59,63 @@ def __init__(self, state: State):
         """
         self.state = state
 
-    def _render_file(self, path: str, content: str, mode: int) -> None:
+    def _render_file(self, path: Path, content: str, mode: int) -> None:
         """Write a content rendered from a template to a file.
 
         Args:
-            path: the path to the file.
+            path: Path object to the file.
             content: the data to be written to the file.
             mode: access permission mask applied to the
               file using chmod (e.g. 0o640).
+
+        Raises:
+            FileRenderError: if interaction with the filesystem fails
         """
-        with open(path, "w+", encoding="utf-8") as file:
-            file.write(content)
-        # Ensure correct permissions are set on the file.
-        os.chmod(path, mode)
         try:
+            path.write_text(content)
+            os.chmod(path, mode)
             # Get the uid/gid for the root user (running the service).
             # TODO: the user running the jenkins agent is currently root
             # we should replace this by defining a dedicated user in the apt package
             u = pwd.getpwnam("root")
             # Set the correct ownership for the file.
             os.chown(path, uid=u.pw_uid, gid=u.pw_gid)
-        except KeyError:
-            # Ignore non existing user error when it wasn't created yet.
-            pass
+        except Exception as exc:
```

Comment:
Shouldn't we be more specific in catching this error ?

Reply:
I replaced this with more specific errors